src/marqo/tensor_search/tensor_search.py

```diff
@@ -1807,7 +1807,55 @@ def sort_chunks(docs: dict) -> List:
     return sorted(as_list, key=lambda x: x["chunks"][0]["_score"], reverse=True)
 
 
+def check_index_health(config: Config, index_name: str) -> dict:
+    """Checks the health of an index
```

Comment:
A question about coding practice here. There is a lot of duplicated code with the old API. Would it be good to wrap the duplicated code to avoid duplication here? Or should I keep the duplication so we can remove the old API easily in the future?

Reply:
You can think about breaking out some repeated code into its own dedicated function.

For example:
```
def calculate_status (...) -> dict:
```

This also makes the logic contained in the function much easier to test